src/mip/mip_types.h

```diff
@@ -34,6 +34,21 @@ typedef int_least16_t remaining_count;
 
 typedef timestamp_type timeout_type;
 
+
+#ifndef MIP_DIAG_INC
```

Comment:
I would prefer no indentation with`#if` or `#ifndef` since we don't use it anywhere else. Also, do we need to check if `MIP_DIAG_INC` is defined? I don't see it being able to be defined by the user easily, and if the include guards work right, it should never get redefined

Reply:
There are no other cases of nested #if, other than where we have #ifdef __cplusplus around most of the file. I think indenting the nested directives when they're near each other makes it easier to read. But maybe that's just me?

I was considering the case where the user may want diagnostic counters to wrap around rather than saturate at the limits, hence the #ifndef MIP_DIAG_INC. But I agree it makes little sense to allow the user to define it on the cmd line for that situation. I will change it.